src/internal/connector/graph/service_helper.go

```diff
@@ -94,31 +94,51 @@ func (handler *LoggingMiddleware) Intercept(
 	}
 
 	if (resp.StatusCode / 100) == 2 {
-		return resp, err
-	}
-
-	// special case for supportability: log all throttling cases.
-	if resp.StatusCode == http.StatusTooManyRequests {
-		logger.Ctx(ctx).Infow("graph api throttling", "method", req.Method, "url", req.URL)
-	}
+		if logger.DebugAPI || os.Getenv(logGraphRequestsEnvKey) != "" {
+			metadata := []any{
+				"idx", middlewareIndex,
+				"method", req.Method,
+				"status", resp.Status,
+				"statusCode", resp.StatusCode,
+				"requestLen", req.ContentLength,
+				"url", req.URL,
+			}
+
+			logger.Ctx(ctx).Debugw("2xx graph api resp", metadata...)
+		}
 
-	if resp.StatusCode != http.StatusTooManyRequests && (resp.StatusCode/100) != 2 {
-		logger.Ctx(ctx).Infow("graph api error", "method", req.Method, "url", req.URL)
+		return resp, err
 	}
 
 	if logger.DebugAPI || os.Getenv(logGraphRequestsEnvKey) != "" {
 		respDump, _ := httputil.DumpResponse(resp, true)
 
 		metadata := []any{
+			"idx", middlewareIndex,
 			"method", req.Method,
-			"url", req.URL,
-			"requestLen", req.ContentLength,
 			"status", resp.Status,
 			"statusCode", resp.StatusCode,
-			"request", string(respDump),
+			"requestLen", req.ContentLength,
+			"url", req.URL,
+			"response", string(respDump),
 		}
 
 		logger.Ctx(ctx).Errorw("non-2xx graph api response", metadata...)
+	} else {
+		// special case for supportability: log all throttling cases.
+		if resp.StatusCode == http.StatusTooManyRequests {
+			logger.Ctx(ctx).Infow("graph api throttling", "method", req.Method, "url", req.URL)
+		}
+
+		if resp.StatusCode != http.StatusTooManyRequests && (resp.StatusCode/100) != 2 {
+			logger.Ctx(ctx).Infow("graph api error", "status", resp.Status, "method", req.Method, "url", req.URL)
+		}
```

Comment:
having this in the else block will only log the starting message when we're not running in debug mode. This can make tracking down throttling more difficult. Maybe we should just always log this even if it means we'll see repeated messages in the log? At least then the "graph api throttling" part will be stable

Reply:
I'm not too concerned.  The debugging mode isn't representative of any of our standard use cases, and will automatically capture throttling responses as part of the non-2xx log.